client/src/components/GoalCreator.tsx

```diff
@@ -13,9 +13,11 @@ function GoalCreator({ moduleID, addGoal }: GoalCreatorProps) {
 
   async function handleModuleCreation() {
     try {
+      // TODO: We need a way for the user to pick goal type before this point.
       const response = await post("/goal/add", {
```

Comment:
You could move your API call up a level to the page and then pass the data down to the component. If you ever wanted to reuse this component, that would allow it to truly be reusable